### PR TITLE
Removed creating of unused objects

### PR DESCRIPTION
**Pull request by tbw777**
To effictive use of memory and for clean code.

### CODE REVIEW

File: enterprise/j2ee.ejbcore/src/org/netbeans/modules/j2ee/ejbcore/action/HomeMethodGenerator.java [hunk 1]
```diff
@@ -73,7 +73,7 @@ public void generate(MethodModel methodModel, boolean generateLocal, boolean gen
 
         // remote interface
         if (generateRemote && remoteHome != null) {
-            List<String> exceptions = exceptions = new ArrayList<String>(methodModel.getExceptions());
+            List<String> exceptions = new ArrayList<String>(methodModel.getExceptions());
```
Comment (BradWalker) [inline enterprise/j2ee.ejbcore/src/org/netbeans/modules/j2ee/ejbcore/action/HomeMethodGenerator.java:76]: Since we're here.. I would mkae this into ArrayList<>..

File: nbi/engine/src/org/netbeans/installer/utils/system/launchers/impl/CommonLauncher.java [hunk 2]
```diff
@@ -95,7 +95,7 @@ protected long addData(FileOutputStream fos, File file, Progress progress, long
     
     //add rnd data
     protected void addData(FileOutputStream fos) throws IOException {
-        double rand = Math.random() * Byte.MAX_VALUE;
+        //double rand = Math.random() * Byte.MAX_VALUE;
```
Comment (BradWalker) [inline nbi/engine/src/org/netbeans/installer/utils/system/launchers/impl/CommonLauncher.java:98]: Remove both comments.. Dead code elimination..

File: platform/api.visual/src/org/netbeans/modules/visual/anchor/DefaultAnchorShapeResolver.java [hunk 3]
```diff
@@ -62,7 +62,7 @@ public int getEndLocation()
         else
         {
 //            Rectangle bounds = attachedWidget.getBounds();
-            Rectangle bounds = new Rectangle();
+            Rectangle bounds;
```
Comment (BradWalker) [inline platform/api.visual/src/org/netbeans/modules/visual/anchor/DefaultAnchorShapeResolver.java:65]: I would split the declaration and assignment into:
Widget sourceWidget;

And then refactor the code to remove the use of bounds..

File: platform/masterfs/src/org/netbeans/modules/masterfs/ProvidedExtensionsProxy.java [hunk 4]
```diff
@@ -79,7 +79,7 @@ public ProvidedExtensions.DeleteHandler getDeleteHandler(final File f) {
     }
     
     public ProvidedExtensions.IOHandler getRenameHandler(final File from, final String newName) {
-        final File to = new File(from.getParentFile(), newName);
+//        final File to = new File(from.getParentFile(), newName);
```
Comment (BradWalker) [inline platform/masterfs/src/org/netbeans/modules/masterfs/ProvidedExtensionsProxy.java:82]: If it's not used, remove it..

File: platform/o.n.swing.plaf/src/org/netbeans/swing/plaf/nimbus/NimbusLFCustoms.java [hunk 5]
```diff
@@ -45,7 +45,7 @@ public Object[] createLookAndFeelCustomizationKeysAndValues() {
         }
 
         //XXX fetch the custom font size here instead
-        Font controlFont = new Font("Dialog", Font.PLAIN, fontsize); //NOI18N
+        //Font controlFont = new Font("Dialog", Font.PLAIN, fontsize); //NOI18N
```
Comment (BradWalker) [inline platform/o.n.swing.plaf/src/org/netbeans/swing/plaf/nimbus/NimbusLFCustoms.java:48]: If it's not used, remove it..

File: platform/o.n.swing.tabcontrol/demosrc/org/netbeans/swing/tabcontrol/demo/TestFrame.java [hunk 6]
```diff
@@ -189,7 +189,7 @@ public void actionPerformed(ActionEvent ae) {
         
         jb9.addActionListener(new ActionListener() {
             public void actionPerformed(ActionEvent ae) {
-                int[] idxs = new int [] { 1, 3, 6, 8};
+                //int[] idxs = new int [] { 1, 3, 6, 8};
```
Comment (BradWalker) [inline platform/o.n.swing.tabcontrol/demosrc/org/netbeans/swing/tabcontrol/demo/TestFrame.java:192]: Not used, remove it..

File: platform/openide.filesystems/src/org/openide/filesystems/AbstractFolder.java [hunk 7]
```diff
@@ -820,7 +820,7 @@ protected final void refreshFolder(
                 fire && (added == null) && (removed == null) && !getFileSystem().isReadOnly() &&
                     !(this instanceof MultiFileObject)
             ) {
-                Set<String> nameFilter = nameFilter = new HashSet<String>();
+                Set<String> nameFilter = new HashSet<String>();
```
Comment (BradWalker) [inline platform/openide.filesystems/src/org/openide/filesystems/AbstractFolder.java:823]: Change to HastSet<>

File: profiler/profiler.heapwalker/src/org/netbeans/modules/profiler/heapwalk/ui/ReferencesBrowserControllerUI.java [hunk 8]
```diff
@@ -354,9 +354,9 @@ public void run() {
             });
             // --- 
         } else {
-            Object[] shortPath = new Object[length];
-            System.arraycopy(path, 0, shortPath, 0, length);
-            final TreePath p = new TreePath(shortPath);
+//            Object[] shortPath = new Object[length];
+//            System.arraycopy(path, 0, shortPath, 0, length);
+//            final TreePath p = new TreePath(shortPath);
             SwingUtilities.invokeLater(new Runnable() {
```
Comment (BradWalker) [inline profiler/profiler.heapwalker/src/org/netbeans/modules/profiler/heapwalk/ui/ReferencesBrowserControllerUI.java:360]: Not used, remove it..

**Review by BradWalker — CHANGES_REQUESTED** [review]
Please update from my comments..

File: nbi/engine/src/org/netbeans/installer/utils/system/launchers/impl/CommonLauncher.java [hunk 9]
```diff
@@ -95,8 +95,6 @@ protected long addData(FileOutputStream fos, File file, Progress progress, long
     
     //add rnd data
     protected void addData(FileOutputStream fos) throws IOException {
-        double rand = Math.random() * Byte.MAX_VALUE;
-        //fos.write(new byte[] {(byte)rand});
         fos.write(new byte[] {'#'});
     }
```
Comment (mbien) [inline nbi/engine/src/org/netbeans/installer/utils/system/launchers/impl/CommonLauncher.java:99]: i don't know why this is commented out but the method does not make sense without it

File: platform/api.visual/src/org/netbeans/modules/visual/anchor/DefaultAnchorShapeResolver.java [hunk 10]
```diff
@@ -49,33 +49,23 @@ public DefaultAnchorShapeResolver(ConnectionWidget connection,
...
+            if (attachedEnd == ConnectionEnd.SOURCE) {
+                sourceWidget = getConnection().getSourceAnchor().getRelatedWidget();
+            } else {
+                sourceWidget = getConnection().getTargetAnchor().getRelatedWidget();
             }
-            retVal = getLocation(bounds);
+            retVal = getLocation(sourceWidget.getBounds());
         }
-        
+
```
Comment (mbien) [inline platform/api.visual/src/org/netbeans/modules/visual/anchor/DefaultAnchorShapeResolver.java:68]: please don't change the formatting of big blocks. This feels unnecessary and obfuscates the actual change here.

File: nbi/engine/src/org/netbeans/installer/utils/system/launchers/impl/CommonLauncher.java (hunk 9 above)
Comment (tbw777) [reply nbi/engine/src/org/netbeans/installer/utils/system/launchers/impl/CommonLauncher.java:99]: Because rand  unused and redundant multiplication operation

File: platform/api.visual/src/org/netbeans/modules/visual/anchor/DefaultAnchorShapeResolver.java (hunk 10 above)
Comment (tbw777) [reply platform/api.visual/src/org/netbeans/modules/visual/anchor/DefaultAnchorShapeResolver.java]: reverted formatting below

File: nbi/engine/src/org/netbeans/installer/utils/system/launchers/impl/CommonLauncher.java (hunk 9 above)
Comment (mbien) [reply nbi/engine/src/org/netbeans/installer/utils/system/launchers/impl/CommonLauncher.java:99]: I see that it is unused. I am wondering why. because the comment above the says "add rnd data". So we have to find out what the method is actually supposed to do before removing anything.

File: platform/o.n.swing.plaf/src/org/netbeans/swing/plaf/nimbus/NimbusLFCustoms.java [hunk 11]
```diff
@@ -44,8 +44,6 @@ public Object[] createLookAndFeelCustomizationKeysAndValues() {
             fontsize = in.intValue();
         }
 
-        //XXX fetch the custom font size here instead
-        Font controlFont = new Font("Dialog", Font.PLAIN, fontsize); //NOI18N
```
Comment (mbien) [inline platform/o.n.swing.plaf/src/org/netbeans/swing/plaf/nimbus/NimbusLFCustoms.java:48]: this belongs to the commented out section below. Comment it out too or revert the change.

File: nbi/engine/src/org/netbeans/installer/utils/system/launchers/impl/CommonLauncher.java (hunk 9 above)
Comment (mbien) [reply nbi/engine/src/org/netbeans/installer/utils/system/launchers/impl/CommonLauncher.java:99]: best approach would be to simply leave it as is or comment it out. Maybe someone wants to use the other version in future.

File: platform/api.visual/src/org/netbeans/modules/visual/anchor/DefaultAnchorShapeResolver.java [hunk 12]
```diff
@@ -49,33 +49,30 @@ public DefaultAnchorShapeResolver(ConnectionWidget connection,
...
-                Widget sourceWidget = getConnection().getTargetAnchor().getRelatedWidget();
-                bounds = sourceWidget.getBounds();
+                sourceWidget = getConnection().getTargetAnchor().getRelatedWidget();
             }
-            retVal = getLocation(bounds);
+            retVal = getLocation(sourceWidget.getBounds());
         }
-        
+
         return retVal;
```
Comment (mbien) [inline platform/api.visual/src/org/netbeans/modules/visual/anchor/DefaultAnchorShapeResolver.java:76]: looking at the code again: this almost looks like a bug that it takes the `sourceWidget` bounds in both cases.

Lets not obfuscate this.

simply change
`Rectangle bounds = new Rectangle();`
into
`Rectangle bounds;`
if you want to get rid of the additional object and lets leave everything else as is.

File: platform/api.visual/src/org/netbeans/modules/visual/graph/layout/hierarchicalsupport/LayeredGraph.java [hunk 13]
```diff
@@ -168,8 +168,7 @@ public float[] computeLowerBarycenters(int upperLayerIndex) {
         int upperLayerSize = upperLayer.size();
         int lowerLayerSize = lowerLayer.size();
         float lowerBarycenters[] = new float[lowerLayerSize];
-        
-        float[] barycenters = new float[lowerLayerSize];
+
```
Comment (mbien) [inline platform/api.visual/src/org/netbeans/modules/visual/graph/layout/hierarchicalsupport/LayeredGraph.java:171]: please remove the comment in L184 too.

**Review by mbien — CHANGES_REQUESTED** [review]

**Comment by tbw777** [comment]
You are right

**Review by mbien — APPROVED** [review]
thanks. looks good to me.
